### PR TITLE
Provide more information about host sub-statuses

### PR DESCRIPTION
**Pull request by ares**
Added more info about how to add searching by custom statuses

### CODE REVIEW

File: developer_docs/how_to_create_a_plugin.asciidoc [hunk 1]
```diff
@@ -35,7 +35,7 @@ There is a fully working example plugin which you can clone to get quickly
 started. It contains examples of many of the types of behaviour that you
 might want to do in a plugin, such as adding new models, overriding
 views, extending controllers, adding permissions and menu items, and so
-on. The README contains a list of it's current behaviour. To get started
+on. The README contains a list of35806 it's current behaviour. To get started
```
Comment (stejskalleos) [inline developer_docs/how_to_create_a_plugin.asciidoc:38]: Is this intentional?

File: developer_docs/how_to_create_a_plugin.asciidoc [hunk 2]
```diff
@@ -1458,6 +1457,35 @@ my_host.refresh_statuses
 The method refreshes *all* statuses by default, this is usually not what
 you want so provide status for refresh.
 
+For each status, the user should be able to search host by it's possible values.
+Your plugin must extend the `Host::Managed` object. Here is an example of how
+such extension definition could look like
```
Comment (stejskalleos) [inline developer_docs/how_to_create_a_plugin.asciidoc:1462]: Missing `.` or `:`?

File: developer_docs/how_to_create_a_plugin.asciidoc [hunk 3]
```diff
@@ -1458,6 +1457,35 @@ my_host.refresh_statuses
...
+         # association name, it needs to be specified explicitly (as a string). Similarly for the foreign key.
+         has_one :execution_status_object, :class_name => 'HostStatus::ExecutionStatus', :foreign_key => 'host_id'
+         
+         # Then we define the searching, the relation is the association name we define on a line above
+         # :rename key indicates the term user will use to search hosts for a given state, the convention is $feature_status
+         # :complete_value must be a hash with symbolized keys specifying all possible state values, otherwise the autocompletion
+         # would only offer values that are already in the database, however it's not guaranteed that user have hosts covering
+         # the whole set of values, that's why the explicit list is necessary. That way user can easily search of all hosts with
+         # "error" even though no host has such status.
+         scoped_search :relation => :execution_status_object, :on => :status,
```
Comment (stejskalleos) [inline developer_docs/how_to_create_a_plugin.asciidoc:1480]: Nit: Maybe would be worth to mention [scoped_search gem](https://github.com/wvanbergen/scoped_search), so users know where `scoped_search` method came from

**Review by stejskalleos — CHANGES_REQUESTED** [review]
Few nits, otherwise ok

**Comment by ares** [comment]
thanks, replaced by https://github.com/theforeman/foreman/pull/9569 based on my fork, that PR resolves your comments

**Pull request closed without merging**